File: c0821g1-building-management/src/app/service/space/space.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {SpacesList} from '../../model/space/spaces-list';
import {environment} from '../../../environments/environment';
import {Spaces} from '../../model/space/spaces';

const API_URL = 'http://localhost:8080/api';

@Injectable({
  providedIn: 'root'
})
export class SpaceService {

  constructor(private httpClient: HttpClient) {
  }

  saveNewSpace(newSpace: Spaces): Observable<void> {
    return this.httpClient.post<void>(API_URL + '/spaces/create', newSpace);
  }

  findByID(id: number): Observable<Spaces> {
    return this.httpClient.get<Spaces>(API_URL + '/spaces/' + id);
  }


  findAllSpace(): Observable<SpacesList[]> {
    return this.httpClient.get<SpacesList[]>(API_URL + '/spaces/list');
  }

  searchSpace(floorName: string,
              spaceCode: string,
              spaceArea: string,
              spaceTypeName: string,
              spaceStatusName: string): Observable<SpacesList[]> {
    return this.httpClient.get<SpacesList[]>(API_URL + '/spaces/search?floorName=' +
      floorName + '&spaceCode=' +
      spaceCode + '&spaceArea=' +
      spaceArea + '&spaceTypeName=' +
      spaceTypeName + '&spaceStatusName=' +
      spaceStatusName);
  }

  deleteSpaceById(spaceId: number): Observable<SpacesList> {
    return this.httpClient.get<SpacesList>(API_URL + `/spaces/delete/${spaceId}`);
  }

  getSpaceById(spaceId: number): Observable<any> {
    return this.httpClient.get(API_URL + `/spaces/detail/${spaceId}`);
  }
  editSpace(spaceEdit: Spaces): Observable < void > {
    return this.httpClient.patch<void>(API_URL + '/spaces/edit/' + spaceEdit.spaceId, spaceEdit);
  }
}

File: sprint1/src/main/java/com/c0821g1/sprint1/service/SpaceService.java
package com.c0821g1.sprint1.service;

import com.c0821g1.sprint1.dto.SpaceListDTO;
import com.c0821g1.sprint1.entity.space.Spaces;

import java.util.List;

public interface SpaceService {
    List<SpaceListDTO> findAllSpace();

    void deleteById(Integer id);

    List<SpaceListDTO> searchSpace(String floorName, String spaceCode, String spaceArea, String spaceTypeName, String spaceStatusName);

    SpaceListDTO findSpaceById(Integer spaceId);

    void saveNewSpace(Spaces spaces);

    Spaces findById(Integer id);

    void save(Spaces spaces);

    void editSpace(Spaces spaces);

    boolean existsSpaceByCode(String spaceCode);

    public List<Spaces> getAll1();
}